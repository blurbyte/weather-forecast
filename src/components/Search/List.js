import styled from 'styled-components';

const List = styled.ul`
  position: absolute;
  top: 3.9rem;
  right: 0;
  width: 36rem;
  border: 1px solid #222222;
  background-color: #ffffff;
  margin: 0;
  padding: 0;
  cursor: pointer;
`;

export default List;
