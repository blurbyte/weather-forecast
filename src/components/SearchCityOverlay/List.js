import styled from 'styled-components';

const List = styled.ul`
  width: 100%;
  background-color: #ffffff;
  padding: 0;
  margin: 0;
  border-bottom: 1px solid #222222;
  cursor: pointer;
`;

export default List;
