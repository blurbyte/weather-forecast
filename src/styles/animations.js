import { keyframes } from 'styled-components';

export const fadeIn = keyframes`
  from {
    opacity: 0;
    transform: translateY(2rem);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;
