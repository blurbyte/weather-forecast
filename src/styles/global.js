import { injectGlobal } from 'styled-components';

injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Cardo:700&subset=latin-ext');

  html {
    font-size: 62.5%;
  }

  body {
    background: #71d7ed;
    color: #222222;
    font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif;
    line-height: 1.4;
    font-size: 1.4rem;
    font-weight: 400;
    overflow-y: scroll;
    user-select: none;
  }
`;
