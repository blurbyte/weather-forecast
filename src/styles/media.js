// Generates media templates based on device breakpoints

import { css } from 'styled-components';

// Devices breakpoints
export const devices = {
  desktop: 1000,
  tablet: 640,
  phone: 420
};

const media = Object.keys(devices).reduce((acc, label) => {
  acc[label] = (strings, ...interpolations) => css`
    @media (max-width: ${devices[label] / 16}em) {
      ${css(strings, ...interpolations)};
    }
  `;
  return acc;
}, {});

export default media;
