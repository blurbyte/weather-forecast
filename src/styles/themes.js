export const dayTheme = {
  primary: '#222222',
  background: '#71d7ed'
};

export const nightTheme = {
  primary: '#ffffff',
  background: '#3033ba'
};
