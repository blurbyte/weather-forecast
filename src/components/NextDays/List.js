import styled from 'styled-components';

import media from '../../styles/media';

const List = styled.ul`
  display: grid;
  list-style: none;
  padding: 0;
  margin: 0;
  grid-template-columns: repeat(4, 1fr);
  border-top: 1px solid ${props => props.theme.primary};

  ${media.tablet`
    display: flex;
    flex-direction: column;
    padding-top: 1rem;
  `};
`;

export default List;
