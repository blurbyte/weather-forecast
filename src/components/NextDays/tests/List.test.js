import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { dayTheme } from '../../../styles/themes';
import List from '../List';

test('renders correctly', () => {
  const component = renderer.create(<List theme={dayTheme} />);
  expect(component.toJSON()).toMatchSnapshot();
});
