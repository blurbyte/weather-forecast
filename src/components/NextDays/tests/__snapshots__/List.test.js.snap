// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: grid;
  list-style: none;
  padding: 0;
  margin: 0;
  grid-template-columns: repeat(4,1fr);
  border-top: 1px solid #222222;
}

@media (max-width:40em) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-top: 1rem;
  }
}

<ul
  className="c0"
/>
`;
